src/parser/helper/Language.ts

```diff
@@ -102,6 +109,57 @@ export const fileExtensionToLanguage = new Map([
     ["sh", Language.Bash],
 ]);
 
+/**
+ * Maps supported file extensions to the corresponding programming languages.
+ * For case-sensitive file extensions.
+ */
+const caseSensitiveFileExtensionToLanguage = new Map([
+    ["c", Language.C],
+    ["C", Language.CPlusPlus],
+    ["H", Language.CPlusPlus], // lowercase .h has a special treatment
+]);
+
+/**
+ * Estimates the language of a file based upon the file extension and file path.
+ * @param filePath Path to the file, including the file extension.
+ * @param config Configuration to apply.
+ * @param pathModule ONLY FOR TESTING PURPOSES: overrides the platform-specific path module.
+ */
+export function assumeLanguageFromFilePath(
+    filePath: string,
+    config: Configuration,
+    pathModule = path,
+) {
+    const fileExtension: string = getFileExtension(filePath);
+
+    // Handling of the parse .h as C option:
+    if (fileExtension === "h") {
```

Comment:
Let's have an early return here: if (!isHeaderFile(fileExtension)) { return; }

Reply:
In Line 139 there is an else-block which is executed on default, so at least the else-block should not be needed.